Source/RapyutaSimulationPlugins/Private/Drives/RobotVehicleMovementComponent.cpp

```diff
@@ -99,13 +106,13 @@ void URobotVehicleMovementComponent::UpdateOdom(float InDeltaTime)
     AActor* owner = GetOwner();
 
     // position
-    FVector Pos = owner->GetActorLocation() - InitialTransform.GetTranslation();
-    FVector PreviousPos = PreviousTransform.GetTranslation();
+    FVector Pos = InitialTransform.GetRotation().UnrotateVector(owner->GetActorLocation() - InitialTransform.GetTranslation());
+    FVector PreviousPos = PreviousTransform.GetTranslation(); //prev pos without noise
     PreviousTransform.SetTranslation(Pos);
     Pos += PreviousEstimatedPos - PreviousPos + WithNoise * FVector(GaussianRNGPosition(Gen), GaussianRNGPosition(Gen), 0);
 
     FRotator NoiseRot = FRotator(0, 0, WithNoise * GaussianRNGRotation(Gen));
-    FQuat Rot = InitialTransform.GetRotation().Inverse() * owner->GetActorRotation().Quaternion();
+    FQuat Rot =  owner->GetActorRotation().Quaternion() * InitialTransform.GetRotation().Inverse();

```

Comment:
Should this also mean:
```suggestion
    FQuat Rot =  InitialTransform.GetRotation() * owner->GetActorQuat();
```
if it means rotating to the owner's rotation first then applying the initial rotation as in the order noted [here](https://github.com/EpicGames/UnrealEngine/blob/release/Engine/Source/Runtime/Core/Public/Math/Quat.h#L25)?

Reply:
calculating the relative pose from initialframe,
R_w = R_r * R_init
R_r = R_W * R_init ^ -1